@xen-orchestra/proxy/src/app/_debounceResource.js

```diff
@@ -1,6 +1,6 @@
 import Disposable from 'promise-toolbox/Disposable'
 
-export async function debounceResource(pDisposable, hooks, delay = 0) {
+export async function debounceResource(pDisposable, addDisposeListener = dispose => Function.prototype, delay = 0) {
```

Comment:
I feel this API is akward to use, should we provide a factory instead?

```js
const myDebounce = createDebounce()

app.hooks.on('stop', myDebounce.disposeAll())
```

Reply:
How to remove the listener with this API? Please see the `debounceResource` implementation.

Reply:
I've just updated my comment, the code was incorrect.

This pushes the responsibility to remove the listener directly out of `debounce` to the consumer code .

Reply:
How the consumer will know that the `dispose` was called from the function `debounceResource` ?

Reply:
I feel `flushAll` is clearer than `disposeAll`.